src/seamless_communication/models/inference/translator.py

```diff
@@ -74,10 +73,22 @@ def __init__(
             pad_idx=self.text_tokenizer.vocab_info.pad_idx, pad_to_multiple=2
         )
         # Load the vocoder.
-        self.vocoder: Vocoder = load_vocoder_model(vocoder_name_or_card, device=device)
-        self.vocoder.eval()
+        self.vocoder: Vocoder = self.load_model_for_inference(
+            load_vocoder_model, vocoder_name_or_card, device, torch.float32
```

Comment:
Is there a specific reason why we use fp32 for Vocoder?

Reply:
Considering that we instantiate the model in fp16, I think it might make sense, but @cndn has much more experience with Vocoder, maybe there might be some culprit I am not aware of.

Reply:
The vocoder checkpoint is in torch.float32, it's also evaluated in torch.float32, let me re-evaluate vocoders in torch.float16 and make sure they have parity before changing this.